### ISSUE TITLE
Some questions about Image Segmentation using Digits 5 

### ISSUE THREAD

**Issue by 4575759ww** [NVIDIA/DIGITS#1438]
Hi, I have worked with FCN models for image segmentation using digits 5, my dataset is about UAV remote sensing images, and I labeled them, all images are spilted into 512*512. However, the result is not good, and the network is indiscriminately classifying everything as one class which is crop with 48% accuracy,   and also, I have tested the Sythina dataset, the network has just learnt to achieve 35% accuracy by labeling everything as building.

So how could I deal with this problem, I have read the article "Image Segmentation Using DIGITS 5",but I’m having trouble understanding the part: transfer learning. I have downloaded the FCN-Alexnet caffemodel as pre-trained caffe model, but it doesn't work in increasing accuracy, so If there are other methods to deal with the question by adjusting some parameters or  something else? 

Hope your feedback and advice! Thanks

**Comment by gheinrich** [comment]
Hello, did you follow all the steps in https://github.com/NVIDIA/DIGITS/tree/digits-5.0/examples/semantic-segmentation#model-creation ?

**Comment by 4575759ww** [comment]
hello, @gheinrich. Thank you for your feedback, I have also used the PASCAL VOC dataset for segmezntation before, and the result is good, almost the same as the result in your paper. Then, I created my own dataset about ground Orthographic images which are obtained by UAV (unmanned aerial vehicle), the UAV image is large, so I spilted one image into 81small images with 512*512 size. 

I followed all the steps in the segmentation examples and used the FCN-Alexnet network, Then I test one image which is the original image with size 2048*2048 in consideration of GPU memory, however, everything is labelled as crop, and there are five classes (road, building, vegetation, crop, river) in my dataset.

Also, I have tested Sythina Rand CVPR16 data, everything is labelled as building.

**Comment by gheinrich** [comment]
Did you provide a pre-trained model too?

**Comment by 4575759ww** [comment]
I didn't provide a pre-trained model about the result I said above . But, I also tried provide the pre-trained FCN-Alexnet caffemodel which exist in the digits 5 package, the result was more bad, the loss value of curve is extremely high, and the accuracy value is  close to zero with accuracy curve keeping sparse.

I am not sure that the pre-trained model I used is correct or not, I also use the net_surgery.py script to convolutionalize an ILSVRC2012-trained Alexnet model, then download the bvlc_alexnet.caffemodel, but when I used it as pre-trained model, it created an error and terminated the job.

So what should I do next? Hope your response.

**Comment by 4575759ww** [comment]
@gheinrich, thank you for your remind. I provide a pre-trained model just now, and obtain a result which is not bad, the accuracy value increased to 78%.

And I found something else，when I adjust the shift value of the Power layer in data processing, the result is changing. So I want to know why the original shift value was -116.0,  and what is the meaning of the Power layer? 

Hope your response.

**Comment by gheinrich** [comment]
Hello, we are using a power layer with shift value of -116 to distribute the data approximately evenly around 0. You can remove the Power layer and use mean pixel subtraction in DIGITS UI instead if your dataset has a different mean.

**Comment by 4575759ww** [comment]
@gheinrich , thank you for your response, it helps a lot.

**Comment by 4575759ww** [comment]
@gheinrich ,when I test one image which size is 4096*4096 or 2048*2048 in the last , there is no output image with warning : list index out of range.  when the size of  input source image is 1024*1024, I can obtain the output  inference visualization image , so could you please tell me the reason for these results?

**Comment by gheinrich** [comment]
Do you have as many labels in your `class labels` file as there are feature maps in the final layer of the network?

**Comment by 4575759ww** [comment]
what do you mean about that there are feature maps in the final layer of the network? I'm not very understanding.

I have 14 classes in my label images, and it is the same with the class labels file.

**Comment by gheinrich** [comment]
How many feature maps do you have in the final layer i.e. what is `num_output` set to in `upscore`? If it's greater than the number of classes you have in your class labels file that might explain the `list index out of range` error. I should probably make the error more explicit if that is the case.

**Comment by 4575759ww** [comment]
I get it, thank you for your answer.

**Comment by 4575759ww** [comment]
@gheinrich, hello. I want to show some data and results to you for some questions, however, some images I used need to be kept confidential, so could you please tell me your email address? Thank you.

**Comment by gheinrich** [comment]
If those images are confidential, better not send them to me :-)
